### ISSUE TITLE
Total time and skip idle together may produce empty video files

### ISSUE THREAD

**Issue by McPringle** [zdila/gpx-animator#318]
Test files: [issue-318.zip](https://github.com/zdila/gpx-animator/files/5898315/issue-318.zip)
Bug identified by @willjobs based on #314.

**Comment by McPringle** [comment]
Looks like the source of the error is the calculation of the *speedup* value. In this case we have four tracks. It takes the earliest time of a track point and the latest time of a track point (in milliseconds since January 1st 1970). These are the start and end times for the animation. Both values are subtracted from each other (to get the travel duration) and together with the tail duration and the frames per second the result is the speed of the animation (where 1.0 is realtime):

```
speedup = (maxTime * fps + tailDuration * fps - minTime * fps) / (frames * MS);
```

In case of the attached example files this calculation looks like:

```
(          maxTime * fps + tailDuration * fps -           minTime * fps) / (frames * MS)    = speedup
(1'604'848'148'634 * 30  +    3'600'000 * 30  - 1'537'621'269'397 * 30)  / (   240 * 1'000) = 8'403'809.904625
```

The resulting animation will be generated with more than 8 million times the speed of the real world. This is amazing fast! Especially, if you want to fit everything in 8 seconds. Together with 30 frames per sonds, there are only 240 frames for the whole animation. If you render the animation without skip idle, the resulting video shows only the time between the four trips. The trips itself are extremly fast and take less than one frame. The result is, that the trip is not animated. It just occours on the next frame!

If you aktivate *skip idle*, the parts between the four trips are cut out. But the trips itself are still extremly fast and take less than a frame. Nothing to animate, not a single frame will be in the resulting video.

To validate this, I removed the *total time* and set the *speedup* value fix to 1'000. Together with *skip idle* activated, the resulting video is 21 seconds long. I played around with the *speedup* value and setting it to 2'700, the resulting video is 8 seconds long. Mission accomplished!

**@willjobs, using this information you should be able to render the video and get the result, you are looking for!**

Nevertheless, GPX Animator should consider the *skip idle* option when calculating the *speedup* for a given *total time* in order to provide a correct result. The most complicated part of this fix will be building a good formula to calculate the idle time before the rendering process starts. This will certainly take me some time (help appreciated). This, together with the fact that there is a workaround, leads me to the decision to bring the fix with version 1.7 and not with 1.6.1.

**Comment by willjobs** [comment]
@McPringle, yup, this works as you said! However, the behavior I was actually going for was a bit different (and I guess maybe not what GPXAnimator was intended for?) I wanted all of my tracks to start at the same "time", so that when I click play on the video, it looks like my running routes are "diffusing" through the network of streets, kind of like [video 2](https://imgur.com/Ub3nHh4) in #314.

**Comment by McPringle** [comment]
Hi @willjobs, I forgot to write a comment on your last sentence. GPX Animator reads the time information and synchronizes multiple tracks automatically. If you don't want this behavior, use a little trick: adjust the track time:

![image](https://user-images.githubusercontent.com/1254039/125681073-2a3ca5b5-1f4d-43e1-996f-3f3443be07cf.png)

So you can adjust the time of the track in a way, that each track starts at the same time.

To make this easier, there could be an option to ignore the real-time from the GPX file. I added a feature request: #361

**Comment by McPringle** [comment]
I was working on this issue for some hours but still have no idea, how to solve it. Not to delay the release of GPX Animator any further, I'll remove this issue from the v1.7 milestone. The impact should be minimal because we had only a report from one user so so far. I hope for your understanding and I'm looking for help to solve this issue…
